output/streaming/stdout.go

```diff
@@ -0,0 +1,73 @@
+package streaming
+
+import (
+	"context"
+	"log"
+
+	"github.com/pkg/errors"
+
+	"github.com/cube2222/octosql/execution"
+	"github.com/cube2222/octosql/storage"
+)
+
+type StreamPrinter struct {
+	stateStorage storage.Storage
+	recordSink   execution.IntermediateRecordStore
+	printFn      func(record *execution.Record)
+}
+
+func NewStreamPrinter(stateStorage storage.Storage, recordSink execution.IntermediateRecordStore, printFn func(record *execution.Record)) *StreamPrinter {
+	return &StreamPrinter{
+		stateStorage: stateStorage,
+		recordSink:   recordSink,
+		printFn:      printFn,
+	}
+}
+
+func (sp *StreamPrinter) Run(ctx context.Context) error {
+	for {
+		tx := sp.stateStorage.BeginTransaction()
+		ctx := storage.InjectStateTransaction(ctx, tx)
+
+		rec, err := sp.recordSink.Next(ctx, tx)
+		if err == execution.ErrEndOfStream {
+			err := tx.Commit()
+			if err != nil {
+				log.Println("couldn't commit transaction: ", err)
+				continue
+			}
+			break
+		} else if errors.Cause(err) == execution.ErrNewTransactionRequired {
+			if err := tx.Commit(); err != nil {
```

Comment:
why this Commit is 1liner when `ErrEndOfStream` and `GetErrWaitForChanges` cases use Commit as 2liners?